common/src/main/java/com/microsoft/identity/common/internal/logging/CommonCoreLogger.java

```diff
@@ -0,0 +1,264 @@
+package com.microsoft.identity.common.internal.logging;
+
+import android.os.Build;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.microsoft.identity.common.adal.error.ADALError;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.concurrent.atomic.AtomicReference;
+
+public final class CommonCoreLogger {
+    private static final CommonCoreLogger INSTANCE = new CommonCoreLogger();
+    static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
+
+    // Turn on the VERBOSE level logging by default.
+    private LogLevel mLogLevel  = LogLevel.VERBOSE;
+
+    // Disable to Logcat logging by default.
+    private boolean mLogcatLogEnabled = false;
+
+    // Disable to log PII by default.
+    private boolean mPIIEnabled = false;
+
+    private AtomicReference<ILoggerCallback> mExternalLogger = new AtomicReference<>(null);
+
+    /**
+     * Enum class for LogLevel that the sdk recognizes.
+     */
+    public enum LogLevel {
+        /**
+         * Error level logging.
+         */
+        ERROR,
+        /**
+         * Warn level logging.
+         */
+        WARN,
+        /**
+         * Info level logging.
+         */
+        INFO,
+        /**
+         * Verbose level logging.
+         */
+        VERBOSE
+        // TODO: 3/19/2018 Do we need debug level?
+    }
+
+    /**
+     * @return The single instance of {@link CommonCoreLogger}
+     */
+    public static CommonCoreLogger getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Set the log level for diagnostic purpose. By default, the sdk enables the verbose level logging.
+     *
+     * @param logLevel The {@link LogLevel} to be enabled for the diagnostic logging.
+     */
+    public void setLogLevel(final LogLevel logLevel) {

```

Comment:
This seems like it could harmlessly be a `static` method and a `static` flag. Does exposing these methods as instance methods offer some benefit?

Reply:
The only difference will be the memory allocation time of the `mLogLevel` property. I prefer to keep the `setLogLevel()` non-static and the same calling pattern `CommonCoreLogger.getInstance.setters` with other setters.